sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -106,8 +111,13 @@ private[sql] object Dataset {
     sparkSession.withActive {
       val qe = new QueryExecution(
         sparkSession, logicalPlan, shuffleCleanupMode = shuffleCleanupMode)
-      qe.assertAnalyzed()
-      new Dataset[Row](qe, RowEncoder.encoderFor(qe.analyzed.schema))
+      val encoder = if (qe.isLazyAnalysis) {
+        RowEncoder.encoderFor(new StructType())
```

Comment:
I wonder if we can add a test to trigger this case

Reply:
Every test using outer references will pass this case?